src/uct/ib/rdmacm/rdmacm_ep.c

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2017.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "rdmacm_ep.h"
+
+static UCS_CLASS_INIT_FUNC(uct_rdmacm_ep_t, uct_iface_t *tl_iface,
+                           const ucs_sock_addr_t *sockaddr,
+                           uct_sockaddr_conn_reply_callback_t reply_cb,
+                           void *arg, uint32_t cb_flags,
+                           const void *priv_data, size_t length)
+{
+    uct_rdmacm_iface_t *iface = ucs_derived_of(tl_iface, uct_rdmacm_iface_t);
+    size_t ip_len = ucs_max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN);
+    char *ip_str  = ucs_alloca(ip_len);
+    uct_rdmacm_priv_data_hdr_t hdr;
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &iface->super);
+
+    ucs_assert(iface->is_server == 0);
```

Comment:
Why iface can't be client for one peer and server for other one?

Reply:
when an iface is created (in uct_iface_open) it can be created either as a server or as a client. 
```params->open_mode``` decides this.

Reply:
does it mean that server can send data only to its clients? I.e., i can't create an endpoint connected to other server for proxying or work distribution, it looks like synthetic limitation.

Reply:
a server can communicate only with its clients. but that's the usual mechanism for client-server work flows right? a user can make a certain process both connect to another server (by creating an ep to it) and also open a uct_iface and start listening to connection requests from other processes.

Reply:
I'm not sure that to create another iface would be more optimal for proxy-server, for example. i think, it looks like not needed complication for user, at least. do you have understanding why it's more optimal?
i see only one pros here - locking per iface. but it can be done without this limitation, let's give user choice.

Reply:
using the same iface as server and then as client (to another server) is currently impossible becuase calling rdma_resolve_addr after rdma_listen on the _same_ cm_id won't work.
but after we introduce a pool of cm_ids (instead of having one cm_id per iface) i think this flow will become possible.

Reply:
As we discussed offline, if creating of EPs on the server iface connected to other server ifaces is possible (now or extendable in future), i'm ok with the solution 👍